### ISSUE TITLE
Research ways that currencies are displayed

### ISSUE THREAD

**Issue by BeejLuig** [BeejLuig/currency-converter#1]
Put reference websites below:

**Comment by BeejLuig** [comment]
Potential parsing roadblocks:
- currencies/symbols in separate tags
- duplicate prices in container tags (accessibility feature?)
- Symbols that don't match with our static map -- will mostly affect currencies that use the "$" symbol

Potential solutions:
- https://github.com/mktj/parsecurrency -- can handle parsing with different currency/number combos. Returns a nice object
- For the above symbol issue, we could simply add an attribute to the static currency objects to support alternate symbols. It would be tedious but the number of currencies using "$" isn't that big (~10)
- Look into use case for [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) (yay browser APIs 🙌)

The *currency* sections of the `toLocaleString` docs reference this [official currency list](https://www.currency-iso.org/en/home/tables/table-a1.html). I used this list for the static `currency-codes.json`

**Comment by BeejLuig** [comment]
Looks like we can handle issue #2 from here as well

**Comment by kerrykimrusso** [comment]
> Look into use case for toLocaleString (yay browser APIs 🙌)

The amount of localization in web APIs always astounds me even though it shouldn't.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#Performance

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat

**Comment by BeejLuig** [comment]
Yes, we want to lean on as many web APIs as possible, only using 3rd party libraries if necessary

**Comment by BeejLuig** [comment]
Pretty thorough article on currency localization: https://sites.miis.edu/zzhou/2017/12/19/formatting-currency-by-using-javascript/
